planner/core/rule_eliminate_projection.go

```diff
@@ -137,6 +138,16 @@ func (pe *projectionEliminater) eliminate(p LogicalPlan, replace map[string]*exp
 		}
 	}
 	p.replaceExprColumns(replace)
+	if len(p.Children()) > 0 {
+		if child, ok := p.Children()[0].(*LogicalProjection); isProj && ok {
+			if len(proj.Exprs) == len(child.Exprs) {
+				for i := range proj.Exprs {
+					proj.Exprs[i] = replaceColumnOfExpr(proj.Exprs[i], child)
```

Comment:
Can we merge it with `replaceExprColumns`? It should be a more general optimization.

Reply:
I am not sure if all operators could do this optimization.
So I just add some special judgemen.
I just know`projection->join/apply`,`projection->sort` may do this optimization.

Reply:
It seems all projection except the final one can be eliminate?